api/tests/unit/domain/usecases/handle-training-recommendation_test.js

```diff
@@ -0,0 +1,109 @@
+const handleTrainingRecommendation = require('../../../../lib/domain/usecases/handle-training-recommendation');
+const campaignRepository = require('../../../../lib/infrastructure/repositories/campaign-repository');
+const trainingRepository = require('../../../../lib/infrastructure/repositories/training-repository');
+const userRecommendedTrainingRepository = require('../../../../lib/infrastructure/repositories/user-recommended-training-repository');
+const { expect, sinon, domainBuilder } = require('../../../test-helper');
+
+describe('Unit | UseCase | handle-training-recommendation', function () {
+  let findByCampaignParticipationIdAndLocaleStub;
+  let saveStub;
+
+  beforeEach(function () {
+    findByCampaignParticipationIdAndLocaleStub = sinon.stub(
+      trainingRepository,
+      'findByCampaignParticipationIdAndLocale'
+    );
+    saveStub = sinon.stub(userRecommendedTrainingRepository, 'save').resolves();
+  });
+
+  describe('when assessment is for campaign', function () {
+    describe('when given target-profile is associate to trainings', function () {
+      it('should create user-recommended-training for user', async function () {
+        // given
+        const locale = Symbol('locale');
+        const campaignParticipationId = Symbol('campaign-participation-id');
+        const domainTransaction = Symbol('domain-transaction');
+        const assessment = domainBuilder.buildAssessment.ofTypeCampaign({ campaignParticipationId });
+        const trainings = [domainBuilder.buildTraining(), domainBuilder.buildTraining()];
+        findByCampaignParticipationIdAndLocaleStub.resolves(trainings);
+
+        // when
+        await handleTrainingRecommendation({
+          locale,
+          assessment,
+          campaignRepository,
+          trainingRepository,
+          userRecommendedTrainingRepository,
+          domainTransaction,
+        });
```

Comment:
```suggestion
        await handleTrainingRecommendation({
          locale,
          assessment,
          trainingRepository,
          userRecommendedTrainingRepository,
          domainTransaction,
        });
```

`campaignRepository ` pas utilisé